crates/primitives/src/revm/env.rs

```diff
@@ -123,7 +123,7 @@ pub fn tx_env_with_recovered(transaction: &TransactionSignedEcRecovered) -> TxEn
 
     #[cfg(feature = "optimism")]
     {
-        let mut envelope_buf = Vec::default();
+        let mut envelope_buf = Vec::new();
```

Comment:
Can we instead allocate this with capacity? This is called a lot, so this reallocation may hit perf.

```suggestion
        let mut envelope_buf = Vec::with_capacity(transaction.length_without_header());
```